services/src/main/java/org/fao/geonet/api/users/MeApi.java

```diff
@@ -23,71 +23,125 @@
+            @ApiParam(
+                hidden = true
+            )
+            HttpSession session
+        ) throws Exception {
+
         UserSession userSession = ApiUtils.getUserSession(session);
-        return userSession.isAuthenticated() ?
-            new ResponseEntity<>(new MeResponse()
-                .setId(userSession.getUserId())
-                .setProfile(userSession.getProfile().name())
-                .setUsername(userSession.getUsername())
-                .setName(userSession.getName())
-                .setSurname(userSession.getSurname())
-                .setEmail(userSession.getEmailAddr())
-                .setOrganisation(userSession.getOrganisation()),
-                    HttpStatus.OK) :
-            new ResponseEntity<>(NO_CONTENT);
+
+        if (userSession.isAuthenticated()) {
+
+            MeResponse myInfos = new MeResponse().setId(userSession.getUserId()).setProfile(userSession.getProfile().name())
+                    .setUsername(userSession.getUsername()).setName(userSession.getName()).setSurname(userSession.getSurname())
+                    .setEmail(userSession.getEmailAddr()).setOrganisation(userSession.getOrganisation())
+                    .setAdmin(userSession.getProfile().equals(Profile.Administrator))
+                    .setGroupsWithRegisteredUser(getGroups(userSession, Profile.RegisteredUser))
+                    .setGroupsWithEditor(getGroups(userSession, Profile.Editor))
+                    .setGroupsWithReviewer(getGroups(userSession, Profile.Reviewer))
+                    .setGroupsWithUserAdmin(getGroups(userSession, Profile.UserAdmin));
+
+            return new ResponseEntity<>(myInfos, HttpStatus.OK);
+        }
+
+        return new ResponseEntity<>(NO_CONTENT);
+    }
+
+    /**
+     * Retrieves a user's groups.
+     *
+     * @param includingSystemGroups if true, also returns the system groups ('GUEST', 'intranet', 'all')
+     * @param all if true returns all the groups, even those the user doesn't belongs to
+     */
+    private List<Integer> getGroups(UserSession session, Profile profile) throws SQLException {
+        ApplicationContext applicationContext = ApplicationContextHolder.get();
+        final GroupRepository groupRepository = applicationContext.getBean(GroupRepository.class);
+        final UserGroupRepository userGroupRepository = applicationContext.getBean(UserGroupRepository.class);
+        final Sort sort = SortUtils.createSort(Group_.id);
+
+        Specifications<UserGroup> spec = Specifications.where(UserGroupSpecs.hasUserId(session.getUserIdAsInt()));
+        // you're no Administrator
+        // retrieve your groups
+        if (profile != null) {
+            spec = spec.and(UserGroupSpecs.hasProfile(profile));
+        }
+        Set<Integer> ids = new HashSet<>(userGroupRepository.findGroupIds(spec));
+
+        // retrieve all groups and filter to only user one
+        List<Group> groups = groupRepository.findAll(null, sort);
```

Comment:
Not sure if this is required, why are retrieved the groups if finally are returned the group ids only that are also in `ids` set?